web/packages/teleport/src/services/integrations/types.ts

```diff
@@ -185,14 +186,21 @@ export type PluginStatus<D = any> = {
   details?: D;
 };
 
-export type PluginSpec =
-  | PluginOktaSpec
-  | PluginSlackSpec
-  | PluginMattermostSpec
-  | PluginOpsgenieSpec
-  | PluginDatadogSpec
-  | PluginEmailSpec
-  | PluginMsTeamsSpec;
+export type PluginNameToSpec = {
+  okta: PluginOktaSpec;
+  slack: PluginSlackSpec;
+  mattermost: PluginMattermostSpec;
+  opsgenie: PluginOpsgenieSpec;
+  datadog: PluginDatadogSpec;
+  email: PluginEmailSpec;
+  msteams: PluginMsTeamsSpec;
+  [key: string]: any;
+};
+
+export type PluginNameToDetails = {
+  okta: PluginStatusOkta;
+  [key: string]: any;
```

Comment:
do we need the `[key: string]: any;`?

Reply:
I added these for using w/ some `pluginsService` funcs, so specifying one of those `name`s will result in the return being properly typed... Since the default for the `Plugin` type's `Spec`/`Status` fields with no generics provided is `any` I figured I'd just keep that the same. Could remove it though and be a bit stricter.

Reply:
+1 on removing here and above unless absolutely necessary.